Source/Scene/Cesium3DTileBatchTable.js

```diff
@@ -341,32 +471,266 @@ define([
+        var featuresLength = this.featuresLength;
+        //>>includeStart('debug', pragmas.debug);
+        if (!defined(batchId) || (batchId < 0) || (batchId > featuresLength)) {
+            throw new DeveloperError('batchId is required and between zero and featuresLength - 1 (' + featuresLength - + ').');
+        }
+        if (!defined(className)) {
+            throw new DeveloperError('className is required.');
+        }
+        //>>includeEnd('debug');
+
+        // PERFORMANCE_IDEA : cache results in the ancestor classes to speed up this check if this area becomes a hotspot
+        var hierarchy = this._batchTableHierarchy;
+        if (!defined(hierarchy)) {
+            return false;
+        }
+
+        // PERFORMANCE_IDEA : treat class names as integers for faster comparisons
+        var result = traverseHierarchy(hierarchy, batchId, function(hierarchy, instanceIndex) {
+            var classId = hierarchy.classIds[instanceIndex];
+            var instanceClass = hierarchy.classes[classId];
+            if (instanceClass.name === className) {
+                return true;
+            }
+        });
+        return defined(result);
+    };
+
+    Cesium3DTileBatchTable.prototype.isExactClass = function(batchId, className) {
         //>>includeStart('debug', pragmas.debug);
+        if (!defined(className)) {
+            throw new DeveloperError('className is required.');
+        }
+        //>>includeEnd('debug');
+
+        return (this.getClassName(batchId) === className);
+    };
+
+    Cesium3DTileBatchTable.prototype.getClassName = function(batchId) {
+        var featuresLength = this.featuresLength;
+        //>>includeStart('debug', pragmas.debug);
+        if (!defined(batchId) || (batchId < 0) || (batchId > featuresLength)) {
+            throw new DeveloperError('batchId is required and between zero and featuresLength - 1 (' + featuresLength - + ').');
+        }
+        //>>includeEnd('debug');
+
+        var hierarchy = this._batchTableHierarchy;
+        if (!defined(hierarchy)) {
+            return undefined;
+        }
+        var classId = hierarchy.classIds[batchId];
+        var instanceClass = hierarchy.classes[classId];
+        return instanceClass.name;
+    };
+
+    Cesium3DTileBatchTable.prototype.hasProperty = function(batchId, name) {
+        var featuresLength = this.featuresLength;
+        //>>includeStart('debug', pragmas.debug);
+        if (!defined(batchId) || (batchId < 0) || (batchId > featuresLength)) {
+            throw new DeveloperError('batchId is required and between zero and featuresLength - 1 (' + featuresLength - + ').');
+        }
         if (!defined(name)) {
             throw new DeveloperError('name is required.');
         }
         //>>includeEnd('debug');
 
         var json = this.batchTableJson;
-        return defined(json) && defined(json[name]);
+        if (defined(json) && defined(json[name])) {
```

Comment:
Style but why not write this entire block as
```javascript
return (defined(json) && defined(json[name]) || (defined(this._batchTableHierarchy) && hasPropertyInHierarchy(this, batchId, name));
```
Or at least change the last block to just:
```javascript
return (defined(this._batchTableHierarchy) && hasPropertyInHierarchy(this, batchId, name));
```